src/main/java/org/codehaus/mojo/flatten/FlattenMojo.java

```diff
@@ -802,11 +809,20 @@ private ModelBuildingRequest createModelBuildingRequest(File pomFile) {
                 context,
                 project.getRemoteProjectRepositories(),
                 getReactorModelsFromSession());
-        Properties userProperties = this.session.getUserProperties();
-        List<String> activeProfiles = this.session.getRequest().getActiveProfiles();
+        Properties userAndActiveExternalProfilesProperties = new Properties();
+        userAndActiveExternalProfilesProperties.putAll(this.session.getUserProperties());
+        settings.getProfiles().stream()
+                .filter(p -> settings.getActiveProfiles().contains(p.getId()))
+                .forEach(
+                        activeProfile -> userAndActiveExternalProfilesProperties.putAll(activeProfile.getProperties()));
+
+        List<String> activeProfiles = Stream.concat(
+                        this.session.getRequest().getActiveProfiles().stream(),
+                        this.settings.getActiveProfiles().stream())
+                .collect(Collectors.toList());
 
         return new DefaultModelBuildingRequest()
-                .setUserProperties(userProperties)
+                .setUserProperties(userAndActiveExternalProfilesProperties)
```

Comment:
There is also method `DefaultModelBuildingRequest.setInactiveProfileIds`, we shoud add

```
new DefaultModelBuildingRequest()
....
  .setInactiveProfileIds(session.getRequest().getInactiveProfiles())
```

Reply:
Done 👍